File: Angular/src/app/shared/models/Order.ts
import { City } from '../enums/City.enum';
import { OrderStatus } from '../enums/OrderStatus.enum';
import { Customer } from './Customer';
import { Product } from './Product';

export interface Order {
  id: number;
  price: number;
  orderTime: Date;
  city: City;
  address: string;
  products: Product[];
  customer: Customer;
  customerId?: number;
  orderStatus: OrderStatus;
}

File: asp.net/Entities/Order.cs
using System.Collections.Generic;
using Utilities.Enums;

namespace Entities
{
    public class Order
    {
        public Order()
        {
            Products = new List<Product>();
        }
        public int Id { get; set; }
        public double Price { get; set; }
        public City City { get; set; }
        public string Address { get; set; }
        public List<Product> Products { get; set; }
        public int? CustomerId { get; set; }
        public Customer Customer { get; set; }
        public OrderStatus OrderStatus { get; set; }
    }
}
